src/uwtools/rocoto.py

```diff
@@ -26,6 +48,67 @@ def _add_jobname(tree: dict) -> None:
 
 
 # Public functions
+def realize_rocoto_xml(
+    input_yaml: str,
+    rendered_output: str,
+) -> bool:  # pragma: no cover
+    """
+    Main entry point.
+
+    :param input_yaml: Path to YAML input file.
+    :param input_template: Path to input template file.
+    :param rendered_output: Path to write rendered XML file.
+    :param schema_file: Path to schema file.
+    """
+    values = YAMLConfig(input_yaml)
+    tasks = values["tasks"]
+    if isinstance(tasks, dict):
+        _add_jobname(tasks)
+
+    # Validate the YAML.
+    if uwtools.config.validator.validate_yaml(
+        config_file=input_yaml, schema_file=str(_rocoto_schema)
+    ):
+        # Render the template.
+        write_rocoto_xml(
+            input_yaml=input_yaml,
+            input_template=str(_rocoto_template),
+            rendered_output=str(rendered_output),
+        )
+        # Validate the XML.
+        if validate_rocoto_xml(input_xml=str(rendered_output), schema_file=str(_rocoto_template)):
+            # If no issues were detected, report success.
+            return True
+        logging.error("Rocoto validation errors identified in %s", rendered_output)
+        return False
+    logging.error("YAML validation errors identified in %s", input_yaml)
+    return False
+
+
+def validate_rocoto_xml(input_xml: str, schema_file: str) -> bool:
+    """
+    Main entry point.
+
+    :param input_XML: Path to rendered XML file.
+    :param schema_file: Path to schema file.
+    """
+
+    # Validate the XML.
+    with open(schema_file, "r", encoding="utf-8") as f:
+        schema = etree.RelaxNG(etree.parse(f))
+    tree = etree.parse(input_xml)
+    success = schema.validate(tree)
+
+    # Store validation errors in the main log.
+    errors = str(etree.RelaxNG.error_log).split("\n")
+    log_method = logging.debug if len(errors) else logging.info
+    log_method("%s Rocoto validation error%s found", len(errors), "" if len(errors) == 1 else "s")
+    for line in errors:
+        logging.debug(line)
```

Comment:
The logging of these errors should be tied to the `--verbose` flag.

Reply:
So I left it as implicit-verbose with `logging.debug()` since we don't actually directly call args outside of the cli anymore, but there should be a way to pass it down, I'll think about it

Reply:
If `--verbose` is specified on the command line, then the logger will accept `logging,.debug()` messages and emit them to `stderr`; if not, it will not. So, I don't think there's a need to pass `args.verbose` down: I think it's only needed to configure the logger, which then handles would-be conditional behavior (to log or not to log) based on the level set.

But do we really want to force the user to say `--verbose` to see the validation errors? If they don't say `--verbose` and see `88 Rocoto validation errors found`, is that really useful without the extra info? If the behavior of our YAML validation against JSON Schema to show the errors without the `--verbose` flag? I'm not sure, but I thought so... If so, should we be consistent here?

Reply:
@christinaholtNOAA In that 2nd paragraph, are you saying that's what we _have_ or that's what we _want_?

My $.02 is that it is what we _have_ but *not* what we _want_. I think we'd want to always report the errors to the user. By way of comparison, in `uwtools.config.validator.validate_yaml()`, when we're validating the UW YAML with our `.jsonschema` schema, we call `logging.error()` when validation errors are encountered -- not `logging.debug()`.

Reply:
That's fair - do we even need a --verbose flag then?

Reply:
I'd vote for keeping `--verbose` 1. for consistency, as it's currently present in all modes (I think), and 2. because at any point someone might want to add some `logging.debug()` messages, in which case it would be convenient for the plumbing to be in place.